python/common/org/python/datetime/Date.java

```diff
@@ -0,0 +1,416 @@
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object __ne__(org.python.Object x) {
+        if (x instanceof Date) {
+            return org.python.types.Bool.getBool(!(this.equals((Date) x)));
+        }
+        else {
+            throw new org.python.exceptions.TypeError("incompatible datatype("+x.typeName()+")");
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object __gt__(org.python.Object x) {
+        if (x instanceof Date) {
+            return org.python.types.Bool.getBool(!(this.lessThan((Date) x)) && !(this.equals((Date) x)));
+        }
+        else {
+            throw new org.python.exceptions.TypeError("incompatible datatype("+x.typeName()+")");
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object __lt__(org.python.Object x) {
+        if (x instanceof Date) {
+            return org.python.types.Bool.getBool(this.lessThan((Date) x));
+        }
+        else {
+            throw new org.python.exceptions.TypeError("incompatible datatype("+x.typeName()+")");
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object __ge__(org.python.Object x) {
+        if (x instanceof Date) {
+            return org.python.types.Bool.getBool(!(this.lessThan((Date) x)));
+        }
+        else {
+            throw new org.python.exceptions.TypeError("incompatible datatype("+x.typeName()+")");
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object __le__(org.python.Object x) {
+        if (x instanceof Date) {
+            return org.python.types.Bool.getBool(this.lessThan((Date) x) || this.equals((Date) x));
+        }
+        else {
+            throw new org.python.exceptions.TypeError("incompatible datatype("+x.typeName()+")");
+        }
+    }
+
+    @org.python.Method(__doc__ = "")
+    public org.python.Object replace(java.util.Map<java.lang.String, org.python.Object> kwargs) {
+
+        if (kwargs.size() > 3) {
+            int val = kwargs.size();
+            throw new org.python.exceptions.TypeError("function takes at most 3 arguments (" + val + " given)");
+        }
+        if (kwargs.size() == 0) {
+            throw new org.python.exceptions.TypeError("function takes at least one argument");
+        }
+        if (kwargs.get("year") != null){
+            this.year = kwargs.get("year");
+        }
+        if (kwargs.get("month") != null){
+            this.month = kwargs.get("month");
+        }
+        if (kwargs.get("day") != null){
+            this.day = kwargs.get("day");
+        }
+
+        if ((this.year instanceof org.python.types.Int) && (this.month instanceof org.python.types.Int) && (this.day instanceof org.python.types.Int)) {
+
+            if (python.datetime.MINYEAR.value <= ((org.python.types.Int) this.year).value && ((org.python.types.Int) this.year).value <= python.datetime.MAXYEAR.value) {
```

Comment:
Some lines throughout the file are too long according to the coding convention. For exampel this line is 170 characters.